src/main/java/cz/cvut/kbss/termit/service/business/SparqlExternalVocabularyService.java

```
@@ -0,0 +1,236 @@
+
+    private static final Logger LOG = LoggerFactory.getLogger(SparqlExternalVocabularyService.class);
+
+    private final Configuration config;
+    private final VocabularyRepositoryService repositoryService;
+
+    private final AccessControlListService aclService;
+    private ApplicationEventPublisher eventPublisher;
+
+    private static final String LIST_AVAILABLE_VOCABULARIES_QUERY = "import/listAvailableVocabularies.rq";
+    private static final String EXPORT_FULL_VOCABULARY_QUERY = "import/exportFullVocabulary.rq";
+
+    public SparqlExternalVocabularyService(VocabularyRepositoryService repositoryService,
+                                           AccessControlListService aclService,
+                                           Configuration config) {
+        this.repositoryService = repositoryService;
+        this.aclService = aclService;
+        this.config = config;
+    }
+    
+/**
+ * Sends a SPARQL query to fetch list of available vocabularies.
+ * 
+ * @return list of available vocabulary information or null if connection failed
+ */
+    @Override
+    public List<RdfsResource> getAvailableVocabularies() {
+
+        List<RdfsResource> response;
+        try {
+            SPARQLRepository sparqlRepo = initSparqlRepository();
+
+            try (RepositoryConnection conn = sparqlRepo.getConnection()) {
+                String sparqlQuery = Utils.loadQuery(LIST_AVAILABLE_VOCABULARIES_QUERY);
+                TupleQuery query = conn.prepareTupleQuery(sparqlQuery);
+
+                TupleQueryResult result = query.evaluate();
+                response = extractListOfAvailableVocabularies(result);
+
+            } catch (QueryEvaluationException e) {
+                LOG.error(e.getMessage());
+                response = List.of();
+            } finally {
+                sparqlRepo.shutDown();
+            }
+        } catch (RepositoryException ex) {
+            response = List.of();
```

Swallowed exception without reporting it.